Miscellaneous features and fixes

brbsix opened the pull request:
Hi, I had a few changes to submit that you may or may not be interested in.

Most importantly, I noticed an issue when generating multiple build directories (e.g. `dpkg-repack --generate PROGRAM1 PROGRAM2 PROGRAM3`). It moves files for all of the programs into one build directory `./dpkg-repack-$$`, so it appears there is a need for unique build directories. I used the format `./$packagename-$random` though perhaps you'd prefer some other format.

I also noticed that control files had an extra newline appended (i.e. there appeared to be 2 blank lines at the bottom). So I removed the final array push `push @control, "\n";`. There may have been a good reason for that line (perhaps under some other circumstances?), but removing it seemed to fix the issue.

Lastly I want to make repacked archives as close to original as possible (which necessitates making the "repacked by dpkg-repack" control file Description tagline optional). I noticed this was also #788628 in the wishlist at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=788628. I added the option as `--no-tag`.

guillemj commented:
Hi!

Ok, as mentioned off-site I pushed what I had locally, plus few other changes on top. I'm still slightly uncomfortable with apparently pseudonymous contributions, but if the patches are not too big, or not copyrightable and there's no way around that, then I guess I'll concede.

Please rebase and fold the whitespace cleanup patches into one, and remove the trailing dot from the summary commit message line. Do not include unrelated changes, like moving an unrelated option around in the man page or the program. 

I'd probably call the option --[no-]tag-description, so that the corresponding option --[no-]tag-version can be used as well. And an additional --no-tag could be added to disable any tagging at all.

Thanks.

brbsix commented:
Hey, so I rebased the commits as requested. I removed the trailing dots, combined the whitespace cleanup, and fixed the `--no-tag` commit. Also I changed the author/commiter to my legal name, Brian Beffa.

I didn't add `--no-tag` or `--no-tag-version`. Do you plan to append something to the version field in the future to differentiate rebuilt versions from the default?

I looked over the recent changes, looks great. The auto `fakeroot` use definitely helps with that annoyance. Also the use of `File::Temp` module is much safer than `rand`. I would have used `mktemp` in the shell but as I'm not familiar with perl, I didn't think to use it.

guillemj commented:
Hi!

Sorry for the delay, it seems I lost track of this one. :/ I've just merged most of the commits, generalized the --tag stuff, and released 1.42 with these changes. Closing now.
